modules/k8s-operator-crd-support/main.tf

```diff
@@ -0,0 +1,108 @@
+  version               = "~> 0.5"
+  module_depends_on     = [module.k8sop_manifest.wait, data.google_client_config.default.project, data.google_container_cluster.primary.name]
+  additional_components = ["kubectl"]
+
+  create_cmd_entrypoint  = "${path.module}/scripts/kubectl_wrapper.sh"
+  create_cmd_body        = "${local.cluster_endpoint} ${local.token} ${local.cluster_ca_certificate} kubectl apply -f ${local.manifest_path}"
+  destroy_cmd_entrypoint = "${path.module}/scripts/kubectl_wrapper.sh"
+  destroy_cmd_body       = "${local.cluster_endpoint} ${local.token} ${local.cluster_ca_certificate} kubectl delete -f ${local.manifest_path}"
+}
+
+
+resource "tls_private_key" "k8sop_creds" {
+  count     = var.create_ssh_key ? 1 : 0
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+module "k8sop_creds_secret" {
+  source                = "terraform-google-modules/gcloud/google"
+  version               = "~> 0.5"
+  module_depends_on     = [module.k8s_operator.wait]
+  additional_components = ["kubectl"]
+
+  create_cmd_entrypoint  = "${path.module}/scripts/kubectl_wrapper.sh"
+  create_cmd_body        = "${local.cluster_endpoint} ${local.token} ${local.cluster_ca_certificate} kubectl create secret generic ${var.operator_credential_name} -n=${var.operator_credential_namespace} --from-literal=${local.k8sop_creds_secret_key}='${local.private_key}'"
+  destroy_cmd_entrypoint = "${path.module}/scripts/kubectl_wrapper.sh"
+  destroy_cmd_body       = "${local.cluster_endpoint} ${local.token} ${local.cluster_ca_certificate} kubectl delete secret ${var.operator_credential_name} -n=${var.operator_credential_namespace}"
+}
+
+
+data "template_file" "k8sop_config" {
+
+  template = file(var.operator_cr_template_path)
+  vars = {
+    cluster_name             = var.cluster_name
+    sync_repo                = var.sync_repo
+    sync_branch              = var.sync_branch
+    policy_dir               = var.policy_dir
+    secret_type              = var.create_ssh_key ? "ssh" : var.secret_type
+    enable_policy_controller = var.enable_policy_controller ? "true" : "false"
+    install_template_library = var.install_template_library ? "true" : "false"
+  }
+}
+
+module "k8sop_config" {
+  source                = "terraform-google-modules/gcloud/google"
+  version               = "~> 0.5"
+  module_depends_on     = [module.k8s_operator.wait, module.k8sop_creds_secret.wait]
+  additional_components = ["kubectl"]
+
+  create_cmd_entrypoint  = "echo"
+  create_cmd_body        = "'${data.template_file.k8sop_config.rendered}' | ${path.module}/scripts/kubectl_wrapper.sh ${local.cluster_endpoint} ${local.token} ${local.cluster_ca_certificate} kubectl apply -f -"
+  destroy_cmd_entrypoint = "echo"
+  destroy_cmd_body       = "'${data.template_file.k8sop_config.rendered}' | ${path.module}/scripts/kubectl_wrapper.sh ${local.cluster_endpoint} ${local.token} ${local.cluster_ca_certificate} kubectl delete -f -"
+}
+
```

Comment:
Tiny nit - looks like extra whitespace here.

Reply:
indeed, good catch, remove that and one more in acm/outputs.tf in 3c4ea7e. thanks!